### PR TITLE
Stylo:  Store animation properties into gecko's struct

### PR DESCRIPTION
**Pull request by hiikezoe**
This is the counter part of https://bugzilla.mozilla.org/show_bug.cgi?id=1328786

All patches get reviewed by r=@heycam.
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

- [X] These changes do not require tests because it's for stylo.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14879)

### CODE REVIEW

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -1293,6 +1361,87 @@ fn static_assert() {
...
+        self.gecko.mAnimationNameCount = v.0.len() as u32;
+        for (servo, gecko) in v.0.into_iter().zip(self.gecko.mAnimations.iter_mut()) {
+            gecko.mName.assign_utf8(&nsCString::from(servo.0.to_string()));
+        }
+    }
+    pub fn animation_name_at(&self, index: usize)
+        -> longhands::animation_name::computed_value::SingleComputedValue {
+        use Atom;
+        use properties::longhands::animation_name::computed_value::AnimationName;
+        // XXX: Is there any effective ways?
```
Comment (emilio) [inline components/style/properties/gecko.mako.rs]: This kinda sucks, can you file a bug to make Gecko store animation names as `Atom`s?

**Comment by emilio** [comment]
Awesome work!

@bors-servo r=heycam

**Comment by emilio** [comment]
heh @heycam was faster :)

File: components/style/properties/gecko.mako.rs [hunk 2]
```diff
@@ -1293,6 +1361,87 @@ fn static_assert() {
...
+        }
+    }
+    pub fn animation_iteration_count_at(&self, index: usize)
+        -> longhands::animation_iteration_count::computed_value::SingleComputedValue {
+        use properties::longhands::animation_iteration_count::computed_value::AnimationIterationCount;
+
+        if self.gecko.mAnimations[index].mIterationCount.is_infinite() {
+            AnimationIterationCount::Infinite
+        } else {
+            AnimationIterationCount::Number(self.gecko.mAnimations[index].mIterationCount as u32)
```
Comment (hiikezoe) [inline components/style/properties/gecko.mako.rs]: I did forget that #14851 gets already merged into servo/master.  Once this commit gets merged into style tree, we will see compile error.  I needed to drop 'as u32' here.  I will send a new PR soon.  I am sorry for the mistake.

Comment (emilio) [inline components/style/properties/gecko.mako.rs]: Instead of sending new PRs you can amend/fixup and force-push to your branch

Comment (emilio) [inline components/style/properties/gecko.mako.rs]: (But of course a new PR is also fine, I guess)

Comment (hiikezoe) [inline components/style/properties/gecko.mako.rs]: Yes, I meant update this PR.

File: components/style/properties/gecko.mako.rs (hunk 1 above)
Comment (hiikezoe) [inline components/style/properties/gecko.mako.rs]: Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1329169

**Comment by hiikezoe** [comment]
@emilio what should I do to fix the Travis CI errors?  Is there anything I can do?
Thanks!

**Comment by emilio** [comment]
Travis errors are known.

@bors-servo r=heycam

**Pull request merged**